Mathlib/Analysis/Analytic/Prod.lean

```diff
@@ -0,0 +1,97 @@
+/-
+Copyright (c) 2023 Geoffrey Irving. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Geoffrey Irving
+-/
+import Mathlib.Analysis.Analytic.Composition
+import Mathlib.Analysis.Analytic.Linear
+
+/-!
+# Analytic functions on and to product spaces
+-/
+
+noncomputable section
+
+open Topology Classical BigOperators NNReal Filter ENNReal
+
+open Function (uncurry)
+
+open Set Filter Asymptotics
+
+variable {𝕜 E F G H : Type*}
+
+variable [NontriviallyNormedField 𝕜] [NormedAddCommGroup E] [NormedSpace 𝕜 E] [NormedAddCommGroup F]
+  [NormedSpace 𝕜 F] [NormedAddCommGroup G] [NormedSpace 𝕜 G] [NormedAddCommGroup H]
+  [NormedSpace 𝕜 H]
+
+/-- Products of analytic functions are analytic -/
```

Comment:
Is it possible to treat a bilinear/multilinear map rather than product?

Reply:
I don't understand the request, `prod.mk` isn't bilinear!

Reply:
I'm not sure I follow the suggestion, but it seems like this would be best handled by a separate lemma that bilinear maps are analytic composed with this one, and then the chained version.  I'll be doing that in particular to handle the case of multiplication in the next PR.  But it seems like we want this case regardless (and certainly I use it directly all the time downstream).

Reply:
Indeed. I confused `prod` with product.

Reply:
Out of scope for this PR, but coming... https://github.com/girving/ray/blob/01188729d12baffe1bc0d88340223ea6e77ce280/Ray/Analytic/Analytic.lean#L180

Reply:
Another product is possible, which takes two analytic functions E -> F and G -> H and considers the map E x G -> F x H.
Your product is a restriction to the diagonal.
Also, what about things like (Fin n -> E -> F) and its dependent variant (Fin n -> E n -> F n)?

Reply:
The usual way we organize things (e.g for `Continuous`) is to first prove that maps to a product are continuous iff each component is. Then it's easy to deduce that both notions of product work well. AFAIK there's no reason that it wouldn't work for analyticity (see `ContinuousMultilinearMap.prod`)

And indeed the same scheme works for both `Prod` and (finite? I'm not even sure we need it) `Pi` (see `ContinuousMultilinearMap.pi`).

Reply:
Regarding naming I think this is right though. We use `prod` to refer to the "diagonal" product (aka applying the universal property) and `prod_map` to refer to the one Antoine describes.